src/common/types/types.cpp

```diff
@@ -391,6 +443,9 @@ static std::string getIncompleteTypeErrMsg(LogicalTypeID id) {
 }
 
 LogicalType::LogicalType(LogicalTypeID typeID) : typeID{typeID}, extraTypeInfo{nullptr} {
+    if (typeID == LogicalTypeID::DECIMAL) {
```

Comment:
move to the switch case below. Maybe we should switch LogicalTypeID now.

Reply:
getPhysicalType takes a LogicalTypeID, which means that either the if case has to be above getPhysicalType, or we have to throw inside getPhysicalType when the ID is DECIMAL